src/fields/field2_128/extend.rs

```diff
@@ -0,0 +1,355 @@
+                fft_butterfly_forward(fft_array, v, recursive_len, twiddle);
+            }
+            bidirectional_fft(power, coset, nodes_count, &mut fft_array[..recursive_len]);
+            for v in 0..nodes_count {
+                fft_butterfly_diagonal(fft_array, v, recursive_len, twiddle);
+            }
+            fft(
+                Direction::Forward,
+                power,
+                coset + recursive_len,
+                &mut fft_array[recursive_len..],
+            );
+        } else {
+            // Inverse FFT: replace evaluations of the polynomial with coefficients
+            fft(
+                Direction::Backward,
+                power,
+                coset,
+                &mut fft_array[..recursive_len],
+            );
+            for v in (nodes_count - recursive_len)..recursive_len {
+                fft_butterfly_diagonal(fft_array, v, recursive_len, twiddle);
+            }
+            bidirectional_fft(
+                power,
+                coset + recursive_len,
+                nodes_count - recursive_len,
+                &mut fft_array[recursive_len..],
+            );
+            for v in 0..(nodes_count - recursive_len) {
+                fft_butterfly_backward(fft_array, v, recursive_len, twiddle);
+            }
+        }
+    }
+}
+
+/// Interpolate a polynomial from the provided nodes, then evaluate it at points
+/// 0..requested_evaluations.
+pub(crate) fn interpolate(nodes: &[Field2_128], requested_evaluations: usize) -> Vec<Field2_128> {
+    // We first run the bidirectional FFT to interpolate the polynomial, then run forward FFTs over
+    // as many coset as are needed to evaluate all the requested points.
+    //
+    // See "Details of Reed-Solomon encoding" in paper section 3.2.
+    //
+    // The FFT must run in an array whose size is a power of two.
+    let fft_size = nodes.len().next_power_of_two();
+    let power = fft_size.ilog2();
+
+    let mut fft_vec = nodes.to_vec();
+    fft_vec.resize(fft_size, Field2_128::ZERO);
+
+    // Run the bidirectional FFT to get context.nodes_len coefficients of the polynomial, then
+    // fft_size - context.nodes_len evaluations of the polynomial in fft_vec.
+    bidirectional_fft(power, 0, nodes.len(), &mut fft_vec);
+
+    let mut out_vec = vec![Field2_128::ZERO; requested_evaluations];
+
+    // Copy the provided evaluations from the nodes to the output
+    out_vec[..nodes.len()].copy_from_slice(nodes);
+
+    // Copy evaluations from the first coset, if any
+    let range = nodes.len()..min(fft_size, requested_evaluations);
+    let fft_vec_evals = &mut fft_vec[range.clone()];
+    out_vec[range].copy_from_slice(fft_vec_evals);
+
+    // Zero out evaluations in fft_vec so we can use it for FFT again
+    fft_vec_evals.fill(Field2_128::ZERO);
+
+    // Use the forward FFT over the remaining cosets, each of size 2^power, to compute the remaining
+    // requested evaluations.
+    for curr_power in (1..).map_while(|coset| {
```

Comment:
I think `curr_power` is a misleading variable name here. Elsewhere, it is a loop variable when looping up to the logarithm of the FFT size. Here instead, it's set to $coset * 2^{power}$, i.e. the start of the evaluation points in the current coset, thus it's coming from a different domain than other variables with the same name. I'd suggest using `start` instead.